### PR TITLE
Refactor: ProposalDetails components NNS agnostic

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Split some cards in ProposalDetails into NNS specific and NNS agnostic.

Preparing for adding Voting for SNS projects.

# Changes

* Change prop in ProposalSummary to be just the string summary.
* Split current ProposalProposerInfoSection into NnsProposalProposerInfoSection and ProposalProposerInfoSection.
* Split current ProposalProposerDataSection into NnsProposalProposerDataSection and ProposalProposerDataSection.
* Split current ProposalProposerActionsEntry into NnsProposalProposerActionsEntry and ProposalProposerActionsEntry.
* Split current ProposalProposerPayloadEntry into NnsProposalProposerPayloadEntry and ProposalProposerPayloadEntry.
* Rename Proposal.svelte to NnsProposal.svelte.

# Tests

* Adapt tests after refactor.

### CODE REVIEW

**Comment by lmuntaner** [comment]
@peterpeterparker @mstrasinskis please review

There are no UI changes.

File: frontend/src/lib/components/proposal-detail/ProposalProposerPayloadEntry.svelte [hunk 1]
```diff
@@ -1,43 +1,17 @@
...
-      proposalId,
-    });
-  }
-
-  let nnsFunctionKey: string | undefined;
-  $: nnsFunctionKey = getNnsFunctionKey(proposal);
 </script>
 
-{#if nnsFunctionKey !== undefined && proposalId !== undefined}
+{#if payload !== undefined && payload !== null}
```
Comment (mstrasinskis) [inline frontend/src/lib/components/proposal-detail/ProposalProposerPayloadEntry.svelte:14]: nitpick: nonNullish()

**Review by mstrasinskis — APPROVED** [review]
LGTM, thank you!

File: frontend/src/lib/components/proposal-detail/ProposalProposerPayloadEntry.svelte (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/components/proposal-detail/ProposalProposerPayloadEntry.svelte:14]: Thanks for the comment. I realized this was not showing the payload with Skeleton when it should.

I changed the logic.

File: frontend/src/lib/components/proposal-detail/ProposalProposerPayloadEntry.svelte [hunk 2]
```diff
@@ -1,64 +1,40 @@
...
-        <div class="json" data-tid="json-wrapper">
-          <Json json={expandedPayload} />
-        </div>
-      {:else}
-        <SkeletonText />
-        <SkeletonText />
-        <SkeletonText />
-      {/if}
-    </div>
+<div class="content-cell-island">
```
Comment (lmuntaner) [inline frontend/src/lib/components/proposal-detail/ProposalProposerPayloadEntry.svelte:17]: @mstrasinskis I changed the logic and moved part to the parent.

Here is about showing the loading state when payload is being fetched.

File: frontend/src/lib/components/proposal-detail/NnsProposalProposerPayloadEntry.svelte [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    nnsFunctionKey !== undefined &&
+    !$proposalPayloadsStore.has(proposalId)
+  ) {
+    loadProposalPayload({
+      proposalId,
+    });
+  }
+</script>
+
+<!-- Only proposals with nnsFunctionKey and proposalId have payload -->
```
Comment (lmuntaner) [inline frontend/src/lib/components/proposal-detail/NnsProposalProposerPayloadEntry.svelte:33]: Only render the box if there might be a payload.

File: frontend/src/lib/components/proposal-detail/ProposalProposerPayloadEntry.svelte [hunk 4]
```diff
@@ -1,64 +1,40 @@
...
   import { SkeletonText } from "@dfinity/gix-components";
-  import type { Proposal } from "@dfinity/nns";
-  import { getNnsFunctionKey } from "$lib/utils/proposals.utils";
   import { expandObject, isNullish } from "$lib/utils/utils";
 
-  export let proposalId: ProposalId | undefined;
-  export let proposal: Proposal | undefined;
-
-  let payload: object | undefined | null;
+  // `undefined` means that the payload is not loaded yet
```
Comment (lmuntaner) [inline frontend/src/lib/components/proposal-detail/ProposalProposerPayloadEntry.svelte:7]: @mstrasinskis is this correct?

File: frontend/src/lib/components/proposal-detail/ProposalProposerPayloadEntry.svelte [hunk 5]
```diff
@@ -1,64 +1,40 @@
...
+  // `null` means that the payload is `null`
```
Comment (mstrasinskis) [inline frontend/src/lib/components/proposal-detail/ProposalProposerPayloadEntry.svelte:8]:
```suggestion
  // `null` means that the payload was not found
```

File: frontend/src/lib/components/proposal-detail/ProposalProposerPayloadEntry.svelte (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/proposal-detail/ProposalProposerPayloadEntry.svelte:17]: Thanks, lgtm

**Review by mstrasinskis — APPROVED** [review]

**Pull request merged**